### ISSUE TITLE
Multiple menu bars and fields

### ISSUE THREAD

**Issue by jayarathina** [UniTime/unitime#50]
I installed UniTime. But i am getting multiple menu bars during login (ten to be exact). The footer also has the `Version 4.4.114 built on Tue, 25 Jun 2019` to 10 times.

![localhost_8080_UniTime_login do](https://user-images.githubusercontent.com/1886779/61888301-e9edc980-af20-11e9-92e1-48e1889c6199.png)

Some sections of the page (after loging) in also repeats itself.
![localhost_8080_UniTime_gwt jsp_page=listSolutions](https://user-images.githubusercontent.com/1886779/61888440-1efa1c00-af21-11e9-9a4b-69b41514b049.png)

Some fields inside also replicates itself. (see date field in the below image)
![localhost_8080_UniTime_sessionEdit do](https://user-images.githubusercontent.com/1886779/61888534-4650e900-af21-11e9-89e3-eb4ae0de80de.png)

What could be the reason?

**Comment by jayarathina** [comment]
**Log file (if needed)** (from _Tomcat 8.5\logs_)

> [07/25/19 21:00:28] INFO   XmlWebApplicationContext -> Closing WebApplicationContext for namespace 'action-servlet': startup date [Thu Jul 25 18:14:34 IST 2019]; parent: Root WebApplicationContext
[07/25/19 21:00:28] INFO   XmlWebApplicationContext -> Closing Root WebApplicationContext: startup date [Thu Jul 25 18:14:15 IST 2019]; root of context hierarchy
[07/25/19 21:00:28] INFO   DefaultLifecycleProcessor -> Stopping beans in phase 1073741823
[07/25/19 21:00:44] INFO   afterPropertiesSet ->  - Initializing Hibernate ... 
[07/25/19 21:00:45] INFO   Version -> HCANN000001: Hibernate Commons Annotations {4.0.5.Final}
[07/25/19 21:00:45] INFO   Version -> HHH000412: Hibernate Core {4.3.11.Final}
[07/25/19 21:00:45] INFO   ConnectionProviderInitiator -> HHH000130: Instantiating explicit connection provider: org.unitime.commons.hibernate.connection.LoggingDBCPConnectionProvider
[07/25/19 21:00:45] INFO   LoggingDBCPConnectionProvider -> Database connection pool logging is enabled.
[07/25/19 21:00:45] INFO   Dialect -> HHH000400: Using dialect: org.hibernate.dialect.MySQLInnoDBDialect
[07/25/19 21:00:45] INFO   ASTQueryTranslatorFactory -> HHH000397: Using ASTQueryTranslatorFactory
[07/25/19 21:00:45] INFO   UpdateTimestampsCache -> HHH000250: Starting update timestamps cache at region: org.hibernate.cache.spi.UpdateTimestampsCache
[07/25/19 21:00:46] INFO   StandardQueryCache -> HHH000248: Starting query cache at region: org.hibernate.cache.internal.StandardQueryCache
[07/25/19 21:00:46] INFO   run -> InfoCache cleanup thread started.
[07/25/19 21:00:49] INFO   SessionFactoryRegistry -> HHH000094: Bound factory to JNDI name: unitime:hibernate/SessionFactory
[07/25/19 21:00:49] INFO   TransactionFactoryInitiator -> HHH000399: Using default transaction strategy (direct JDBC transactions)
[07/25/19 21:00:49] INFO   <init> -> Reading file:/C:/Program Files/Apache Software Foundation/Tomcat 8.5/webapps/UniTime/WEB-INF/lib/timetable.jar!/dbupdate.xml ...
[07/25/19 21:00:49] INFO   DatabaseUpdate -> Current UniTime database version: 220
[07/25/19 21:00:49] INFO   DatabaseUpdate -> New UniTime database version: 220
[07/25/19 21:00:49] INFO   afterPropertiesSet ->  - Creating Message Log Appender ... 
[07/25/19 21:00:49] INFO   afterPropertiesSet ->  - Initializing Room Availability Service ... 
[07/25/19 21:00:49] INFO   afterPropertiesSet ->  - Cleaning Logs ...
[07/25/19 21:00:49] INFO   afterPropertiesSet ->  - Starting Event Expiration Service ...
[07/25/19 21:00:49] INFO   afterPropertiesSet -> ******* UniTime 4.4.114 build on Tue, 25 Jun 2019 initialized successfully *******
[07/25/19 21:00:49] INFO   EventExpirationService -> Event expiration service started.
[07/25/19 21:00:49] INFO   EventExpirationService -> Checking for expired events ...
[07/25/19 21:00:49] INFO   CourseSolverContainer -> Restore folder: C:\Program Files\Apache Software Foundation\Tomcat 8.5\data\unitime\restore
[07/25/19 21:00:49] INFO   CourseSolverContainer -> Solver passivation thread started.
[07/25/19 21:00:49] INFO   ExaminationSolverContainer -> Solver passivation thread started.
[07/25/19 21:00:49] INFO   StudentSolverContainer -> Solver passivation thread started.
[07/25/19 21:00:49] INFO   InstructorSchedulingContainer -> Solver passivation thread started.
[07/25/19 21:00:49] INFO   OnlineStudentSchedulingContainer -> Student Sectioning Service is starting up ...
[07/25/19 21:00:49] INFO   AbstractSolverServer -> Solver server is up and running.
[07/25/19 21:00:49] INFO   updater[generic] -> Generic updater started.
[07/25/19 21:00:49] INFO   OnlineSectioningLogger -> Online Sectioning Logger is up.
[07/25/19 21:00:49] INFO   DefaultSpringSecurityContextSource ->  URL 'ldap://null', root DN is ''
[07/25/19 21:00:49] INFO   AbstractContextSource -> Property 'userDn' not set - anonymous context will be used for read-write operations
[07/25/19 21:00:52] INFO   MenuBackend -> Reading menu from file:/C:/Program Files/Apache Software Foundation/Tomcat 8.5/webapps/UniTime/WEB-INF/lib/timetable.jar!/menu.xml ...
[07/25/19 21:00:52] INFO   MenuBackend -> Reading custom menu from file:/C:/Program Files/Apache Software Foundation/Tomcat 8.5/webapps/UniTime/WEB-INF/lib/timetable.jar!/menu-custom.xml ...
[07/25/19 21:00:52] INFO   TaskExecutorService -> Task executor service started.
[07/25/19 21:00:52] INFO   JtaTransactionManager -> Using JTA UserTransaction: Transaction: unknown
[07/25/19 21:00:52] INFO   JtaTransactionManager -> Using JTA TransactionManager: Transaction: unknown
[07/25/19 21:00:53] INFO   ThreadPoolTaskScheduler -> Initializing ExecutorService  'taskScheduler'
[07/25/19 21:00:53] INFO   DefaultSecurityFilterChain -> Creating filter chain: org.springframework.security.web.util.matcher.AnyRequestMatcher@1, [org.springframework.security.web.context.SecurityContextPersistenceFilter@6a616c73, org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter@63dc8205, org.springframework.security.web.header.HeaderWriterFilter@38ba2ddf, org.springframework.security.web.authentication.logout.LogoutFilter@3179b28c, org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter@501a5224, org.springframework.security.web.authentication.www.BasicAuthenticationFilter@57d12e88, org.springframework.security.web.savedrequest.RequestCacheAwareFilter@3e6c3578, org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter@42fd1deb, org.unitime.timetable.spring.security.UniTimeAnonymousAuthenticationFilter@4074fce2, org.springframework.security.web.session.SessionManagementFilter@1e8f2f0a, org.springframework.security.web.access.ExceptionTranslationFilter@41970e56, org.springframework.security.web.access.intercept.FilterSecurityInterceptor@6baf51ca]
[07/25/19 21:00:53] INFO   DefaultFilterChainValidator -> Checking whether login URL '/login.do' is accessible with your configuration
[07/25/19 21:00:53] INFO   DefaultLifecycleProcessor -> Starting beans in phase 0
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {transformer} as a subscriber to the 'data-import-file' channel
[07/25/19 21:00:53] INFO   DirectChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.data-import-file' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started org.springframework.integration.config.ConsumerEndpointFactoryBean#0
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {transformer} as a subscriber to the 'data-import-file-enriched' channel
[07/25/19 21:00:53] INFO   DirectChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.data-import-file-enriched' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started org.springframework.integration.config.ConsumerEndpointFactoryBean#1
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {service-activator} as a subscriber to the 'data-import-xml' channel
[07/25/19 21:00:53] INFO   DirectChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.data-import-xml' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started org.springframework.integration.config.ConsumerEndpointFactoryBean#2
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {message-handler} as a subscriber to the 'data-import-log' channel
[07/25/19 21:00:53] INFO   DirectChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.data-import-log' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started org.springframework.integration.config.ConsumerEndpointFactoryBean#3
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {transformer} as a subscriber to the 'data-import-failure' channel
[07/25/19 21:00:53] INFO   DirectChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.data-import-failure' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started org.springframework.integration.config.ConsumerEndpointFactoryBean#4
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {transformer} as a subscriber to the 'data-import-failure-enriched' channel
[07/25/19 21:00:53] INFO   DirectChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.data-import-failure-enriched' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started org.springframework.integration.config.ConsumerEndpointFactoryBean#5
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {message-handler} as a subscriber to the 'data-import-failure-log' channel
[07/25/19 21:00:53] INFO   DirectChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.data-import-failure-log' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started org.springframework.integration.config.ConsumerEndpointFactoryBean#6
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {service-activator} as a subscriber to the 'event-inbound-emails' channel
[07/25/19 21:00:53] INFO   DirectChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.event-inbound-emails' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started org.springframework.integration.config.ConsumerEndpointFactoryBean#7
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> Adding {logging-channel-adapter:_org.springframework.integration.errorLogger} as a subscriber to the 'errorChannel' channel
[07/25/19 21:00:53] INFO   PublishSubscribeChannel -> Channel 'org.springframework.web.context.WebApplicationContext:/UniTime.errorChannel' has 1 subscriber(s).
[07/25/19 21:00:53] INFO   EventDrivenConsumer -> started _org.springframework.integration.errorLogger
[07/25/19 21:00:53] INFO   DefaultLifecycleProcessor -> Starting beans in phase 1073741823
[07/25/19 21:00:53] INFO   SourcePollingChannelAdapter -> started org.springframework.integration.config.SourcePollingChannelAdapterFactoryBean#0
[07/25/19 21:00:53] INFO   ContextLoader -> Root WebApplicationContext: initialization completed in 11829 ms
[07/25/19 21:00:54] INFO   ActionServlet -> Loading chain catalog from jar:file:/C:/Program%20Files/Apache%20Software%20Foundation/Tomcat%208.5/webapps/UniTime/WEB-INF/lib/struts-core-1.3.10.jar!/org/apache/struts/chain/chain-config.xml
[07/25/19 21:00:54] INFO   TilesPlugin -> Tiles definition factory loaded for module ''.
[07/25/19 21:00:54] INFO   ContextLoaderPlugIn -> ContextLoaderPlugIn for Struts ActionServlet 'action, module '': initialization started
[07/25/19 21:00:54] INFO   XmlWebApplicationContext -> Refreshing WebApplicationContext for namespace 'action-servlet': startup date [Thu Jul 25 21:00:54 IST 2019]; parent: Root WebApplicationContext
[07/25/19 21:00:54] INFO   XmlBeanDefinitionReader -> Loading XML bean definitions from ServletContext resource [/WEB-INF/action-servlet.xml]
[07/25/19 21:00:54] INFO   ContextLoaderPlugIn -> Using context class 'org.springframework.web.context.support.XmlWebApplicationContext' for servlet 'action'
[07/25/19 21:00:54] INFO   ContextLoaderPlugIn -> ContextLoaderPlugIn for Struts ActionServlet 'action', module '': initialization completed in 13 ms
[07/25/19 21:00:55] INFO   JProf -> Using wall time.
[07/25/19 21:00:57] INFO   TilesRequestProcessor -> Tiles definition factory found for request processor ''.
[07/25/19 21:05:06] INFO   Registration -> Registration information received.
[07/25/19 21:05:49] INFO   server[Fal2010woebegon] -> Executor is waiting for a new job...
[07/25/19 21:05:49] INFO   server[Fal2010woebegon] -> Config: [
      Comparator.Class: org.cpsolver.ifs.solution.GeneralSolutionComparator
      CourseRequest.SameTimePrecise: true
      Distances.Ellipsoid: WGS84
      Distances.ShortDistanceAccommodationReference: SD
      Enrollment.CanKeepCancelledClass: false
      Extensions.Classes: org.cpsolver.studentsct.extension.DistanceConflict;org.cpsolver.studentsct.extension.TimeOverlapsCounter
      General.BalanceUnlimited: false
      General.SaveBestUnassigned: 0
      General.SaveDefaultProperties: false
      General.StartUpDate: 1564068949773
      Interactive.UpdateCourseRequests: true
      Load.CheckEnabledForScheduling: true
      Load.CheckForNoBatchStatus: true
      Load.IncludeCourseDemands: true
      Load.IncludeLastLikeStudents: false
      Load.RequestGroups: false
      Load.StudentQuery: 
      Load.TweakLimits: false
      Neighbour.BackTrackDepth: 4
      Neighbour.BackTrackTimeout: 5000
      Neighbour.BranchAndBoundMinimizePenalty: false
      Neighbour.BranchAndBoundTimeout: 5000
      Neighbour.Class: org.cpsolver.studentsct.heuristics.StudentSctNeighbourSelection
      Neighbour.MaxValues: 100
      Neighbour.RandomUnassignmentOfProblemStudentProb: 0.9
      Neighbour.RandomUnassignmentProb: 0.5
      Neighbour.SwapStudentsMaxValues: 100
      Neighbour.SwapStudentsTimeout: 5000
      OnlineStudentSectioning.TimesToAvoidHeuristics: true
      OverExpected.Disbalance: 0.100
      OverExpected.Percentage: 1.000
      OverExpected.Rounding: ROUND
      OverExpectedCriterion.Class: org.cpsolver.studentsct.online.expectations.AvoidUnbalancedWhenNoExpectations
      Reservation.CanAssignOverTheLimit: true
      SectionLimit.PreferDummyStudents: true
      Sectioning.KeepInitialAssignments: false
      Student.DummyStudentWeight: 0.01
      StudentLunch.EndStart: 156
      StudentLunch.Length: 6
      StudentLunch.StartSlot: 132
      StudentSct.CBS: true
      StudentSct.ProjectedCourseDemadsClass: None
      StudentSct.StudentDist: true
      StudentSct.TimeOverlaps: true
      StudentSctBasic.Mode: Initial
      StudentSctBasic.WhenFinished: No Action
      StudentWeights.AlternativeRequestFactor: 0.1260
      StudentWeights.AvailabilityFactor: 0.050
      StudentWeights.AvgPenaltyFactor: 0.001
      StudentWeights.BackToBackDistance: 6
      StudentWeights.BackToBackFactor: -0.0010
      StudentWeights.BalancingFactor: 0.0050
      StudentWeights.Class: org.cpsolver.studentsct.online.selection.StudentSchedulingAssistantWeights
      StudentWeights.DistanceConflict: 0.0500
      StudentWeights.FirstAlternative: 0.5010
      StudentWeights.LeftoverSpread: false
      StudentWeights.LunchBreakFactor: 0.0050
      StudentWeights.Mode: Priority
      StudentWeights.MultiCriteria: true
      StudentWeights.NoTimeFactor: 0.050
      StudentWeights.PenaltyFactor: 0.250
      StudentWeights.Perturbation: 0.1000
      StudentWeights.PreferenceFactor: 0.500
      StudentWeights.Priority: 0.5010
      StudentWeights.PriorityWeighting: true
      StudentWeights.ProjectedStudentWeight: 0.0100
      StudentWeights.SameChoice: 0.900
      StudentWeights.SameChoiceFactor: 0.125
      StudentWeights.SameGroup: 0.1000
      StudentWeights.SameNameFactor: 0.014
      StudentWeights.SameRoomsFactor: 0.007
      StudentWeights.SameTime: 0.700
      StudentWeights.SameTimeFactor: 0.070
      StudentWeights.SecondAlternative: 0.2510
      StudentWeights.SelectionFactor: 0.3750
      StudentWeights.ShortDistanceConflict: 0.2000
      StudentWeights.TimeOverlapFactor: 1.0000
      StudentWeights.TimeOverlapMaxLimit: 0.5000
      StudentWeights.TooEarlyFactor: 0.0500
      StudentWeights.TooLateFactor: 0.0250
      StudentWeights.TravelTimeFactor: 0.0010
      StudentWeights.WorkDayFactor: 0.0100
      Suggestions.MaxDepth: 4
      Suggestions.MaxSuggestions: 20
      Suggestions.Timeout: 1000
      Termination.Class: org.cpsolver.ifs.termination.GeneralTerminationCondition
      Termination.StopWhenComplete: true
      Termination.TimeOut: 28800
      TravelTime.MaxTravelGap: 12
      Value.Class: org.cpsolver.studentsct.heuristics.EnrollmentSelection
      Value.WeightConflicts: 1.0
      Value.WeightNrAssignments: 0.0
      Variable.Class: org.cpsolver.ifs.heuristics.GeneralVariableSelection
      WorkDay.EarlySlot: 102
      WorkDay.LateSlot: 210
      WorkDay.WorkDayLimit: 72
      approve-enrollments.ExcludeLockedOfferings: false
      check-assignment.ExcludeLockedOfferings: false
      check-offering.ExcludeLockedOfferings: false
      eligibility.LockOfferings: false
      reject-enrollments.ExcludeLockedOfferings: false
      status-change.LockOfferings: false
      student-email.LockOfferings: false
    ]
[07/25/19 21:05:49] INFO   updater[Fal2010woebegon] -> 2010 Fal (woebegon) updater started.
[07/25/19 21:05:49] INFO   reload-all[Fal2010woebegon] -> Updating course infos and the student sectining model for session 2010 Fal (woebegon)
[07/25/19 21:05:51] INFO   reload-all[Fal2010woebegon] ->   Update of session 2010 Fal (woebegon) done 1.1 seconds.
[07/25/19 21:05:51] INFO   server[Fal2010woebegon] -> Total Allocated 342.99 kB (of 423.98 MB), details: [
      AbstractServer.iConfig: 13.56 kB (103 records)
      InMemoryServer.iCourseForId: 8.57 kB (21 records)
      InMemoryServer.iCourseForName: 11.85 kB (21 records)
      InMemoryServer.iExpectations: 12.55 kB (21 records)
      InMemoryServer.iOfferingRequests: 60.90 kB (21 records)
      InMemoryServer.iOfferingTable: 165.55 kB (21 records)
      InMemoryServer.iStudentTable: 67.66 kB (19 records)
    ]
[07/25/19 21:05:51] INFO   server[Fal2010woebegon] -> Executor is waiting for a new job...
[07/25/19 21:06:24] INFO   Solver -> Using org.cpsolver.ifs.termination.MPPTerminationCondition
[07/25/19 21:06:24] INFO   Solver -> Using org.cpsolver.coursett.heuristics.TimetableComparator
[07/25/19 21:06:24] INFO   Solver -> Using org.cpsolver.ifs.algorithms.SimpleSearch
[07/25/19 21:06:24] INFO   StandardNeighbourSelection -> Using org.cpsolver.coursett.heuristics.PlacementSelection
[07/25/19 21:06:24] INFO   StandardNeighbourSelection -> Using org.cpsolver.coursett.heuristics.LectureSelection
[07/25/19 21:06:24] INFO   StandardNeighbourSelection -> Using org.cpsolver.coursett.heuristics.PlacementSelection
[07/25/19 21:06:24] INFO   StandardNeighbourSelection -> Using org.cpsolver.coursett.heuristics.LectureSelection
[07/25/19 21:06:24] INFO   Solver -> Using org.cpsolver.coursett.heuristics.UniversalPerturbationsCounter
[07/25/19 21:06:24] INFO   Solver -> Using org.cpsolver.ifs.extension.ConflictStatistics
[07/25/19 21:06:24] INFO   Solver -> Using org.cpsolver.ifs.extension.ViolatedInitials
[07/25/19 21:06:24] INFO   Progress -> [Loading input data ...]
[07/25/19 21:06:26] INFO   Progress -> Using room availability that was updated on Thu Jul 25 21:06:26 IST 2019.
[07/25/19 21:06:26] INFO   Progress -> [Student Enrollments Check]
[07/25/19 21:06:26] INFO   Progress -> Model successfully loaded.
[07/25/19 21:06:26] INFO   Progress -> [Awaiting commands ...]
[07/25/19 21:07:17] WARN   PropertyMessageResources ->   Resource messages_en_US.properties Not Found.
[07/25/19 21:07:17] WARN   PropertyMessageResources ->   Resource messages_en.properties Not Found.

**Comment by tomas-muller** [comment]
Such an error has not been reported to us before. I do not see any issues in the log.

What browser are you using? Do you see the same error when you open the [online demo](https://demo.unitime.org/UniTime)? Are you able to reproduce the problem in other browsers?

**Comment by jayarathina** [comment]
> What browser are you using? 

Chrome 75.0.3770.142

> Do you see the same error when you open the [online demo](https://demo.unitime.org/UniTime)? 

online demo is working perfectly

> Are you able to reproduce the problem in other browsers?

Yes. In all browser installed in my system (Opera and Edge), the same error is there. (I tried freshly installed browser and again with all extensions disabled in chrome, the error persists)

Chrome Browser Log is [here](https://pastebin.com/Rw2ZZKKb). This is a screen shot of the chrome browser log:
![browser log](https://user-images.githubusercontent.com/1886779/61938477-d8eb9980-afae-11e9-9a4d-d8421e7dd325.png)

```
WARNING: Uncaught exception: TypeError: $wnd.unitime.runAsyncCallback7 is not a function
java.lang.NullPointerException: TypeError: $wnd.unitime.runAsyncCallback7 is not a function
    at Unknown.anonymous(http://localhost:8080/UniTime/unitime/deferredjs/B1C2752189CADF72E324D3918459ECAC/7.cache.js)
```

Is that of any help?

Is it possible that I deployed it wrongly? (copied some files multiple times?) How to remove UniTime alone and deploy it again? What are the files that I need to delete?

**Comment by tomas-muller** [comment]
I am not able to reproduce the issue using JDK 12.0.2 and Apache Tomcat 8.5.43 in either Chrome 75.0.3770.142 or Firefox 68.0.1. I have tried both the [official released UniTime 4.4.114](https://github.com/UniTime/unitime/releases/tag/v4.4.114) and the latest [nightly build](https://builds.unitime.org/) UniTime 4.4.117 that is currently used in the [online demo](https://demo.unitime.org/UniTime/login.do). But I have a Mac. While most of our production installations (that we have worked closely with in the past) are using Linux-based machines, we have some Windows installations as well which as far as I know are working fine.

The TypeError error is in the [GWT](http://www.gwtproject.org/) client-side JavaScript code that is produced by the GWT compiler from Java code. This makes it hard to debug or understand, especially since about 20 different permutations are being compiled (one for each supported browser and locale).

Are you using the official build of UniTime (downloaded from our [nightly builds](https://builds.unitime.org/) page, GitHub [releases](https://github.com/UniTime/unitime/releases), or SourceForge [releases](https://sourceforge.net/projects/unitime/files/)) or did you build UniTime yourself?

UniTime can be redeployed by placing a new UniTime.war file in Tomcat\webapps (where Tomcat is something like C:\Program Files\Apache Software Foundation\Tomcat 8.5 on Windows). To ensure that any of the cached files are not reused, could you please try to:
1) stop the Apache Tomcat
2) in Tomcat\webapps, remove the UniTime folder and replace the existing UniTime.war with the new one
3) delete the content of Tomcat\work folder
4) start the Apache Tomcat

**Comment by jayarathina** [comment]
> Are you using the official build of UniTime (downloaded from our [nightly builds](https://builds.unitime.org/) page, GitHub [releases](https://github.com/UniTime/unitime/releases), or SourceForge [releases](https://sourceforge.net/projects/unitime/files/)) or did you build UniTime yourself?

I am using the official build of UniTime I downloaded from github. I tried 4.4.117 official nightly build, but same multiple menu problem persists.

**Surprisingly I tried with _UniTime 4.0 (no longer supported)_ version (Version 4.0.49 built on Wed, 24 Feb 2016) at https://builds.unitime.org/ and it worked very well....** :smile: :confetti_ball: I  don't know why.:confused: Is it ok to continue to use it? Are there any serious bug or holes that I should be aware of? (any major changes that are not in current documentation?)

Is there anything else I can do to help you debug 4.4.114 and make it work in my system?

P.S: May I also ask, which is the oldest release of UniTime (if any) that can generate timetable with  classes to availability of professors (i.e., no room constraints)?

**Comment by tomas-muller** [comment]
UniTime 4.0 is using an older version of GWT (2.7.0, same as UniTime 4.1 and 4.2). Later versions use GWT 2.8.2. I am not aware of any serious bug/holes, but there have been some issues with the older versions of GWT on newer browsers or when deploying older versions of UniTime on a newer version of Apache Tomcat.

There have not been many changes in course timetabling since 4.0 except the solver pages having a newer look and a few additional features and some additional constraints in the solver. The page documentation for UniTime 4.0 is available at [http://help40.unitime.org](http://help40.unitime.org).

The course timetabling solver considers instructors and their preferences and requirements from the start. It is possible to ignore room constraints.

The ability to assign instructors to classes (instructor scheduling) has been added in UniTime 4.2.

**Comment by jayarathina** [comment]
Is it possible to update the [eclipse help page](http://help.unitime.org/eclipse) with the latest details? I was trying to install the latest code, but got error probably because i was using GWT 2.7.0 as specified there. Is there anything else that has changed? It would be helpful if u update the screen shots there as GUI have changed too.

**Comment by jayarathina** [comment]
Is it possible to update the [eclipse help page](http://help.unitime.org/eclipse) with the latest details? I was trying to install the latest code, but got error probably because i was using GWT 2.7.0 as specified there. Is there anything else that has changed? It would be helpful if u update the screen shots there as GUI have changed too. If i can successfully run it, then i will try to debug it in my system.

(PS: i closed the issue by mistake, sorry)

**Comment by tomas-muller** [comment]
You need to use the same GWT version as is used in the version of UniTime you are working with. Otherwise, the setup is the same (except, you may need a newer version of eclipse, Apache Tomcat, etc.).

Please note that you do not need Eclipse to check out and build UniTime. See [Building UniTime](http://help.unitime.org/build) for more details.

**Comment by jayarathina** [comment]
I have some question regarding the usage of UniTime (how to do certain things, like changing the start time of the time grid etc.,) where can I ask them?

**Comment by tomas-muller** [comment]
You can send your questions directly to us using the [support@unitime.org](mailto:support@unitime.org) email, or post them on the [unitime-discuss](https://groups.google.com/d/forum/unitime-discuss) forum.

**Comment by jayarathina** [comment]
unitime-discuss link forum says:
> This group either doesn't exist, or you don't have permission to access it. If you're sure this group exists, contact the Owner of the group and ask them to give you access.

Can you please check it out? or do you need to add my email address to the forum so that I can ask question?

**Comment by tomas-muller** [comment]
> Can you please check it out? or do you need to add my email address to the forum so that I can ask question?

Please try again.

**Comment by jvjeuc** [comment]
@tomas-muller 
I just want to notice that I get the exact same issue with multiple items/menu/links etc
Followed the installation guide on a Windows Server
Tomcat 8.5
MySql Server 8

![Skærmbillede 2019-10-03 kl  21 19 26](https://user-images.githubusercontent.com/37147565/66157257-b071c280-e623-11e9-9b5c-d59e5037289c.png)

**Comment by tomas-muller** [comment]
I have still not been able to reproduce this problem. So far, it has only been reported on Windows and only with UniTime 4.4. I was told that UniTime 4.3 is working fine despite the same version of the GWT library (which is responsible for the menus etc.).

**Comment by jvjeuc** [comment]
Yes, it do work with 4.3 :-)

**Comment by tomas-muller** [comment]
This problem should be fixed in UniTime 4.4.130 or later.
